packages/gatsby/cache-dir/api-runner-ssr.js

```diff
@@ -6,10 +6,45 @@
 
 const apis = require(`./api-ssr-docs`)
 
-module.exports = (api, args, defaultReturn) => {
+/**
+ * Some apis should only be implemented once. Given a list of plugins, and an
+ * `api` to check, this will return [] for apis that are implemented 1 or 0 times.
+ *
+ * For apis that have been implemented multiple times, return an array of paths
+ * pointing to the file implementing `api`.
+ *
+ * @param {array} pluginList
+ * @param {string} api
+ */
+const duplicatedApis = (pluginList, api) => {
+  let implementsApi = []
+
+  pluginList.forEach(p => {
+    if (p.plugin[api]) implementsApi.push(p.path)
+  })
+
+  if (implementsApi.length < 2) return [] // no dupes
+  return implementsApi // paths to dupes
+}
+
+// Run the specified api in any plugins that have implemented it
+const apiRunner = ({ api, args, defaultReturn, checkDupes = false }) => {
   if (!apis[api]) {
     console.log(`This API doesn't exist`, api)
   }
+
+  if (checkDupes) {
+    const dupes = duplicatedApis(plugins, api)
+    if (dupes.length > 0) {
+      let m = `\nThe "${api}" api has been implemented multiple times. Only the last implementation will be used.`
+      let m2 = `This is probably an error, see https://github.com/gatsbyjs/gatsby/issues/2005#issuecomment-326787567 for workarounds.`
```

Comment:
This should be a docs page — could you add a new docs page explaining the problem and solutions and link to that instead?

Reply:
Oh, you already said you were going to do this :-)